aten/src/ATen/native/cpu/BinaryOpsKernel.cpp

```diff
@@ -502,6 +502,80 @@ void min_elementwise_kernel(TensorIterator& iter) {
   }
 }
 
+void maximum_kernel(TensorIterator& iter) {
+  if (isIntegralType(iter.dtype(), /*includeBool=*/ true)) {
+    AT_DISPATCH_INTEGRAL_TYPES_AND(at::ScalarType::Bool, iter.dtype(), "maximum_cpu", [&] {
+      cpu_kernel(iter, [](scalar_t a, scalar_t b) -> scalar_t {
+        return a >= b ? a : b;
+      });
+    });
+  } else if (isFloatingType(iter.dtype())) {
+    AT_DISPATCH_FLOATING_TYPES_AND2(at::ScalarType::Half, at::ScalarType::BFloat16, iter.input_dtype(), "maximum_cpu", [&]() {
+      cpu_kernel(iter, [](scalar_t a, scalar_t b) -> scalar_t {
+        if (_isnan(a)) {
+          return a;
+        }
+        if (_isnan(b)) {
+          return b;
+        }
+        return a >= b ? a : b;
+      });
+    });
+  } else {
+    AT_DISPATCH_COMPLEX_TYPES(iter.dtype(), "maximum_cpu", [&] {
+      cpu_kernel(iter, [](scalar_t a, scalar_t b) -> scalar_t {
+        if (_isnan(a.real()) || _isnan(a.imag())) {
```

Comment:
I think isnan works on complex, see:

https://github.com/pytorch/pytorch/blob/2f840b1662b487d5551d7230f8eb4d57645cfff5/aten/src/ATen/native/TensorCompare.cpp#L80

It's documented as doing so:

https://pytorch.org/docs/master/generated/torch.isnan.html?highlight=isnan#torch.isnan

This should save you from looking at the real and imaginary parts separately.

Reply:
However... (see above)

Reply:
Thanks for the detailed explanation of `isnan` function.
As complex numbers are not supported, the complex type dispatch codes are removed in the latest commit.